plugin/pkg/scheduler/algorithm/predicates/metadata_test.go

```diff
@@ -0,0 +1,352 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"fmt"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+	schedulertesting "k8s.io/kubernetes/plugin/pkg/scheduler/testing"
+	"reflect"
+	"testing"
+)
+
+var label1 = map[string]string{
+	"region": "r1",
+	"zone":   "z11",
+}
+var label2 = map[string]string{
+	"region": "r1",
+	"zone":   "z12",
+}
+var label3 = map[string]string{
+	"region": "r2",
+	"zone":   "z21",
+}
+var label4 = map[string]string{
+	"region": "r2",
+	"zone":   "z22",
+}
+
+func equivalentLists(l1, l2 []interface{}) bool {
+	if len(l1) != len(l2) {
+		return false
+	}
+	for _, item1 := range l1 {
+		found := false
+		for _, item2 := range l2 {
+			if item1 == item2 {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+// predicateMetadataEquivalent returns true if the two metadata are equivalent.
+// Note: this function does not compare podRequest.
+func predicateMetadataEquivalent(meta1, meta2 *predicateMetadata) error {
+	if meta1.pod != meta2.pod {
```

Comment:
In this test environment where we pass the same pod pointer to the metadata producers, this is fine. I changed it to DeepEqual anyway to avoid confusion.